cns/middlewares/swiftV2.go

```diff
@@ -34,21 +34,79 @@ type SWIFTv2Middleware struct {
 	Cli client.Client
 }
 
-// ValidateIPConfigsRequest validates if pod is multitenant by checking the pod labels, used in SWIFT V2 scenario.
+func (m *SWIFTv2Middleware) IPConfigsRequestHandlerWrapper(defaultHandler, failureHandler cns.IPConfigsHandlerFunc) cns.IPConfigsHandlerFunc {
+	return func(ctx context.Context, req cns.IPConfigsRequest) (*cns.IPConfigsResponse, error) {
+		podInfo, respCode, message := m.validateIPConfigsRequest(ctx, &req)
+		if respCode != types.Success {
+			return &cns.IPConfigsResponse{
+				Response: cns.Response{
+					ReturnCode: respCode,
+					Message:    message,
+				},
+			}, errors.New("failed to validate ip configs request")
+		}
+		// If the pod is not v2, return the response from the handler
+		if !req.SecondaryInterfacesExist {
```

Comment:
nit: since we're calling defaultHandler anyway - we can move line 53 above the if statement, and just return ipConfigsResp,err on line 50